src/components/ViewResponses/index.js

```diff
@@ -1,25 +1,38 @@
-import { Button, Card, Input, Form, Typography } from "antd";
+import { Button, Input, Form, Typography } from "antd";
 import { useEffect, useState } from "react";
-import { getFormResponses } from "../../utils/nostr";
+import { getFormResponses, getUserKind0s } from "../../utils/nostr";
 import Analytics from "./Analytics";
 import { useParams } from "react-router";
+import Response from "./Responses";
 
-function ViewResponses(props) {
+function ViewResponses() {
   const [nsecState, setNsecState] = useState("");
   const [responses, setResponses] = useState([]);
+  const [userInfo, setUserInfo] = useState({});
   const { nsec } = useParams();
   const { Text } = Typography;
 
   useEffect(() => {
     if (nsec) {
       getResponses(nsec);
     }
-  }, [nsec]);
+  });
 
   async function getResponses(nsecInput) {
     let resp = await getFormResponses(nsecInput);
-    console.log("fetching", resp);
     setResponses(resp);
+    let pubkeysList = resp.map((r) => {
+      return r.pubkey;
+    });
+    let userKind0 = await getUserKind0s(pubkeysList);
+    let userInf = {};
+    userKind0.forEach((kind0) => {
+      userInf[kind0.pubkey] = JSON.parse(kind0.content);
```

Comment:
JSON.parse and JSON.stringify can be handled in the util functions them selves. Will help in typing the objects better

Reply:
This PR is getting bulky, created an issue: https://github.com/abhay-raizada/nostr-forms/issues/28 will fix after this